### ISSUE TITLE
Cannot pass a : character as part of a device name when starting a container

### ISSUE THREAD

**Issue by Nemo157** [docker/cli#742]
**Description**

Cannot pass a `:` character as part of a device name when starting a container.

**Steps to reproduce the issue:**
1. Have a device node containing a `:` character
2. Attempt to run a docker container with this device mounted into it.

**Describe the results you received:**

```sh
> sudo ln -s random /dev/example:device
> docker run --device /dev/example:device alpine sh -c 'head -c 5 /dev/random | base64'
invalid argument "/dev/example:device" for --device: device is not an absolute path
```

**Describe the results you expected:**

Some way to escape the `:` character to successfully mount this device.

**Output of `docker version`:**

```
Docker version 17.10.0-ce, build f4ffd2511c
```

**Comment by thaJeztah** [comment]
Yes, the `--device` (like `--volume`) option uses colons as a separator between options, and uses the [`--device <device on host>:<path-in-container>[:<options>]` format](https://docs.docker.com/engine/reference/commandline/run/#add-host-device-to-container-device)

To resolve this, we'd probably need an additional syntax, similar to what's used for the `--mount` option (or would it make sense to add a `type=device` option to `--mount`? @cpuguy83 ?).

**Comment by cpuguy83** [comment]
`--mount` would be a good fit for this.

**Comment by 0x0elliot** [comment]
seems like this is fixed as well. shouldn't this be closed?

**Comment by thaJeztah** [comment]
I don't think `type=device` has been implemented in the API; https://github.com/moby/moby/blob/8bb58153e75b4310f3ae9ec6d03924aa286fbf20/api/types/mount/mount.go#L7

```go
// Type represents the type of a mount.
type Type string

// Type constants
const (
	// TypeBind is the type for mounting host dir
	TypeBind Type = "bind"
	// TypeVolume is the type for remote storage volumes
	TypeVolume Type = "volume"
	// TypeTmpfs is the type for mounting tmpfs
	TypeTmpfs Type = "tmpfs"
	// TypeNamedPipe is the type for mounting Windows named pipes
	TypeNamedPipe Type = "npipe"
	// TypeCluster is the type for Swarm Cluster Volumes.
	TypeCluster Type = "cluster"
)
```
